canvas_api_client/canvas_api_client/v1_client.py

```diff
@@ -141,17 +153,50 @@ def _get_paginated(self,
                 response.links['next']['url'], headers=headers)
             yield response.json()
 
-    def _format_sis_course_id(self, course_id: str):
+    def _get_flattened(self,
+                       url: str,
+                       headers: RequestHeaders = None,
+                       params: RequestParams = None) -> Iterator[Response]:
+        """
+        Send an API call to the Canvas server with pagination.
+
+        Returns a generator of response objects.
+        """
+        response = self._get(url, headers=headers, params=params)
+        self._check_response_headers_for_pagination(response)
+
+        for item in response.json():
+            yield item
+
+        while 'next' in response.links:
+            response = self._get(
+                response.links['next']['url'],
+                headers=headers
+                )
+
+            for item in response.json():
+                yield item
+
+    def _format_sis_course_id(self, course_id: str,
+                              is_sis_course_id: Optional[bool]):
         """
         Returns request string for querying with a SIS course ID.
         """
-        return "sis_course_id:{}".format(course_id)
+        if is_sis_course_id or self._is_sis_course_id:
+            return "sis_course_id:{}".format(course_id)
+
+        return course_id
 
-    def _format_sis_account_id(self, account_id: str):
+    def _format_sis_account_id(self,
+                               account_id: str,
+                               is_sis_account_id: Optional[bool] = None):
```

Comment:
this looks like a breaking change with the old version of the client. are we sure that there will be nothing downstream that gets effected?

Reply:
i.e. if the feeds depend on this we will need to be sure to fix the calls there..

Reply:
My intention was to make it backwards compatible.  What do you think would break?

Reply:
We have several tests that check to make sure both Canvas and SIS IDs are properly passed through, like `test_get_course_info()` and `test_get_account_roles()`.  Do you think we're missing anything?

Reply:
looks good to me!